src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -321,6 +333,45 @@ protected override AuthenticationTicket AuthenticateCore()
                     _configuration = await Options.ConfigurationManager.GetConfigurationAsync(Context.RequestAborted);
                 }
 
+                if (!string.IsNullOrWhiteSpace(message.Code))
```

Comment:
Switching the order here is problematic in the case of id_token + code. We haven't validated the code w.r.t. id_token (chash) or even validated the token. This should be fixed.